### PR TITLE
Bug 1014440 - Refactored Firefox Accounts panel with AMD pattern.

### PR DESCRIPTION
**Pull request by scottwu**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5a99d047015622c589f226b726d0ae5196eb90ab

File: apps/settings/js/panels/firefox_accounts/firefox_accounts.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+define(function(require) {
+  var Normalizer = require('shared/text_normalizer');
```
Comment (gasolin) [inline apps/settings/js/panels/firefox_accounts/firefox_accounts.js]: name in camelCase please

File: apps/settings/js/panels/firefox_accounts/panel.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      },
+      onBeforeShow: function() {
+        firefoxAccounts.onBeforeShow();
+      },
+      onBeforeHide: function() {
+        firefoxAccounts.onBeforeHide();
+      }
+    });
+  };
+});
```
Comment (gasolin) [inline apps/settings/js/panels/firefox_accounts/panel.js]: Add space at the end line

File: apps/settings/js/panels/firefox_accounts/firefox_accounts.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      refreshStatus: refreshStatus,
+      _onResend: _onResend,
+      _onResendClick: _onResendClick
+    };
+  };
+
+  return function ctor_firefox_accounts() {
+    return new FirefoxAccounts();
+  };
+});
```
Comment (gasolin) [inline apps/settings/js/panels/firefox_accounts/firefox_accounts.js]: Add new blank  line

File: apps/settings/js/panels/firefox_accounts/firefox_accounts.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        fxaHelper.resendVerificationEmail(
+          email, _onResend.bind(null, email), _onFxAccountError);
+      }, _onFxAccountError);
+    };
+
+    var _onResend = function fxa__onResend(email) {
+      return navigator.mozL10n.formatValue('fxa-resend-alert', {
+        email: email
+      }).then((msg) => {
+        window.alert(msg);
```
Comment (gasolin) [inline apps/settings/js/panels/firefox_accounts/firefox_accounts.js]: Use dialogservice alert instead

File: apps/settings/js/panels/firefox_accounts/firefox_accounts.js [hunk 5]
```diff
@@ -0,0 +1,153 @@
+'use strict';
+
+define(function(require) {
+  var Normalizer = require('shared/text_normalizer');
+  var fxaHelper = require('shared/fxa_iac_client');
+
+  var FirefoxAccounts = function() {
```
Comment (gasolin) [inline apps/settings/js/panels/firefox_accounts/firefox_accounts.js]: Wrap with module pattern (Use = {} ) to align with other panels

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0a1d8ad0c55aafd8ffab3f7078fcd4308980e5a9

File: apps/settings/test/unit/mock_dialog_service.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ define(function() {
 
   var DialogService = {
     alert: function() {
+      console.log('>>> alert');
```
Comment (gasolin) [inline apps/settings/test/unit/mock_dialog_service.js]: It can be removed

File: apps/settings/js/panels/firefox_accounts/firefox_accounts.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+    // if e == null, user is logged out.
+    // if e.verified, user is logged in & verified.
+    // if !e.verified, user is logged in & unverified.
+    _onFxAccountStateChange: function fxa_onFxAccountStateChange(e) {
+      var email = e ? normalizer.escapeHTML(e.email) : '';
+      if (!e) {
+        this._hideLoggedInPanel();
+        this._hideUnverifiedPanel();
+        this._showLoggedOutPanel();
```
Comment (gasolin) [inline apps/settings/js/panels/firefox_accounts/firefox_accounts.js]: Could we simplify those 6 methods to something like
```
showLoggedInPanel(false, email)
showUnverifiedPanel(false, email)
showLoggedOutPanel(true)
```

is it simpler or make it more complex?

Comment (gasolin) [inline apps/settings/js/panels/firefox_accounts/firefox_accounts.js]: You can do the refactor in another bug

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8cb2d39e849cecb1db1af73f528dd7f6fc2b6159

File: apps/settings/js/panels/firefox_accounts/firefox_accounts.js (hunk 7 above)
Comment (scottwu) [inline apps/settings/js/panels/firefox_accounts/firefox_accounts.js]: Ok, I'll do that in another patch. Thanks!

File: apps/settings/test/unit/mock_dialog_service.js (hunk 6 above)
Comment (scottwu) [inline apps/settings/test/unit/mock_dialog_service.js]: Thanks for catching this :+1:

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cd404186236fcded3aa22ff167a2d42abd2d8d1f

**Pull request merged**